src/App.tsx

```diff
@@ -1,56 +1,121 @@
-import React from 'react';
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useCallback, useEffect, useState, useRef,
+} from 'react';
+import debounce from 'lodash/debounce';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';
+import classnames from 'classnames';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownMenu } from './components/DropdownMenu';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [delayedQuery, setDelayedQuery] = useState('');
+  const [searchResults, setSearchResults] = useState<Person[]>([]);
+  const [isQueryCleared, setIsQueryCleared] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+  const [activePersonSlug, setActivePersonSlug] = useState<string | null>(null);
 
-  return (
-    <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+  const dropdownRef = useRef<HTMLDivElement>(null);
 
-      <div className="dropdown is-active">
-        <div className="dropdown-trigger">
-          <input
-            type="text"
-            placeholder="Enter a part of the name"
-            className="input"
-          />
-        </div>
+  useEffect(() => {
+    if (isQueryCleared) {
+      setDelayedQuery('');
+      setSearchResults([]);
+      setIsQueryCleared(false);
+      setIsDropdownActive(false);
+    } else {
```

Comment:
Instead of using 'isQueryCleared', you can set 'query' to an empty string and handle this case directly in the useEffect watching 'query'. This way, you can remove 'isQueryCleared' state.

Reply:
Is not only about watching the query, but is also about setting the searchResult to an empty array and the deactivating the dropdown.

I've modified the code to be used for example to search for cities in a shipping component (Ecommerce).

Reply:
@polosanya  Thank you for your input!